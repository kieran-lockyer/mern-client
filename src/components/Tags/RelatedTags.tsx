import React, { Component } from "react";
import { connect } from "react-redux";
import Moment from "react-moment";
import styled from "styled-components";
import baseUrl from "../../api/baseurl";

class RelatedTags extends Component<any, any> {
  renderRelatedTags() {
    if (this.props.tags) {
      return this.props.tags.map((tag, id) => {
        return (
          <div key={id} style={{ display: "grid", gridTemplateRows: "180px" }}>
            <div>
              <img
                src={baseUrl + "/photos/image/" + tag.imageId}
                alt=""
                style={{
                  width: "100%",
                  height: "100%",
                  objectFit: "cover"
                }}
              />
            </div>
            <h3 style={{ lineHeight: "1.5", marginTop: "7px" }}>{tag.label}</h3>
            <p>
              Date: <Moment format="D MMM YYYY">{tag.dateAdded}</Moment>
            </p>
            <p>
              <strong>Type:</strong> {tag.source.type}
            </p>
            <p>
              <strong>Model:</strong> {tag.source.model}
            </p>
            <p>
              <strong>Confidence:</strong> {(tag.confidence * 100).toFixed(2)}
            </p>
          </div>
        );
      });
    }
  }

  render() {
    return <RelatedTagDetails>{this.renderRelatedTags()}</RelatedTagDetails>;
  }
}

const mapStateToProps = state => ({
  tags: state.tags.relatedTags
});

const RelatedTagDetails = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-gap: 2rem;

  &img {
    object-fit: cover;
  }
`;

export default connect(mapStateToProps)(RelatedTags);
