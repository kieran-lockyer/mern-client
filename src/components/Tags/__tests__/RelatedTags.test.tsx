import React from 'react';
import { shallow } from 'enzyme';
import RelatedTags from '../RelatedTags';
import toJson from 'enzyme-to-json';
import configureStore from 'redux-mock-store';

describe('<RelatedTags />', () => {
  describe('render()', () => {
    test('renders the RelatedTags component', () => {
      const component = shallow(<RelatedTags />);
      expect(toJson(component)).toMatchSnapshot();
    });
  });
});