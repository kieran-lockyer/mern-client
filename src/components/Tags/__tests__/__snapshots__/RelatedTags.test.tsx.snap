// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelatedTags /> render() renders the RelatedTags component 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
